lib/tapioca/rbi_merger.rb

```diff
@@ -0,0 +1,34 @@
+# typed: strict
+# frozen_string_literal: true
+
+module Tapioca
+  class RBIMerger
+    class << self
+      extend T::Sig
+
+      sig do
+        params(rbi: RBI::Tree, gem: Gemfile::GemSpec, shell: Thor::Shell::Color)
+          .returns(RBI::Tree)
+      end
+      def merge_with_exported_rbi!(rbi, gem, shell)
+        return rbi unless gem.has_exported_rbi_files?
+
+        copied_rbi = gem.exported_rbi_tree do |conflicts|
+          shell.say("\n #{conflicts.join("\n")}", :yellow)
+        end
+        display_conflict_warnings(rbi, copied_rbi, shell)
+        RBI::Rewriters::Merge.merge_trees(rbi, copied_rbi, keep: RBI::Rewriters::Merge::Keep::LEFT)
+      end
+
+      private
+
+      sig { params(generated_rbi: RBI::Tree, copied_rbi: RBI::Tree, shell: Thor::Shell::Color).void }
+      def display_conflict_warnings(generated_rbi, copied_rbi, shell)
+        rewriter = RBI::Rewriters::Merge.new
+        rewriter.merge(generated_rbi)
+        conflicts = rewriter.merge(copied_rbi)
+        shell.say(conflicts.join("\n"), :yellow) unless conflicts.empty?
```

Comment:
You have access to [both nodes](https://github.com/Shopify/rbi/blob/main/lib/rbi/rewriters/merge_trees.rb#L85-L86) and so their [location](https://github.com/Shopify/rbi/blob/main/lib/rbi/model.rb#L15). I believe we could do better and tell the user which files/lines are creating the conflict.